From: Erik Schmauss <erik.schmauss@intel.com>
Date: Fri, 29 Jun 2018 11:28:12 -0700
Subject: ACPICA: Revert "iASL compiler: allow compilation of externals with
 paths that refer to existing names"
Patch-mainline: v4.19-rc1
Git-commit: a9efdcffd54dc1098a089c3c833eef8d7bee4d77
References: bsc#1117419

Revert commit 3ddd3f6a9410 (ACPICA: iASL compiler: allow compilation
of externals with paths that refer to existing names; upstream
ACPICA commit 9a252114197409290813bee570e9d53c22b99d32).

This was done in order to allow more relaxed usage of ASL external
declarations.

Signed-off-by: Erik Schmauss <erik.schmauss@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 drivers/acpi/acpica/aclocal.h  |    1 -
 drivers/acpi/acpica/nsaccess.c |    6 ------
 2 files changed, 7 deletions(-)

--- a/drivers/acpi/acpica/aclocal.h
+++ b/drivers/acpi/acpica/aclocal.h
@@ -165,7 +165,6 @@ struct acpi_namespace_node {
 #define ANOBJ_EVALUATED                 0x20	/* Set on first evaluation of node */
 #define ANOBJ_ALLOCATED_BUFFER          0x40	/* Method AML buffer is dynamic (install_method) */
 
-#define IMPLICIT_EXTERNAL               0x02	/* iASL only: This object created implicitly via External */
 #define ANOBJ_IS_EXTERNAL               0x08	/* iASL only: This object created via External() */
 #define ANOBJ_METHOD_NO_RETVAL          0x10	/* iASL only: Method has no return value */
 #define ANOBJ_METHOD_SOME_NO_RETVAL     0x20	/* iASL only: Method has at least one return value */
--- a/drivers/acpi/acpica/nsaccess.c
+++ b/drivers/acpi/acpica/nsaccess.c
@@ -608,12 +608,6 @@ acpi_ns_lookup(union acpi_generic_state
 					    this_node->object;
 				}
 			}
-#ifdef ACPI_ASL_COMPILER
-			if (!acpi_gbl_disasm_flag &&
-			    (this_node->flags & ANOBJ_IS_EXTERNAL)) {
-				this_node->flags |= IMPLICIT_EXTERNAL;
-			}
-#endif
 		}
 
 		/* Special handling for the last segment (num_segments == 0) */
